Assets/Scripts/AI/Entities/Colonist.cs

```diff
@@ -1,19 +1,44 @@
 using UnityEngine;
+using WorldAPI.Tasks.Generic;
 
-public class Colonist : AIEntity
+public partial class Colonist : ResourceGatherer<Colonist>
```

Comment:
What does the partial keyword do here?

Reply:
It allows multiple "parts" of a class/interface to be split. The static members `Colonist.Derive()` and `Colonist.New()` is implemented there instead to make it easier to understand the different parts. They can also be merged into a single declaration if necessary.